extensions/2.0/Khronos/KHR_physics_rigid_bodies/README.md

```diff
@@ -0,0 +1,507 @@
+
+JSON is unable to represent infinite values; however, a value of infinity is useful for both `mass` and the components of `inertiaDiagonal`. As zero is an *invalid* value for these properties, a value of zero in `mass` or any of the components of `inertiaDiagnal` should be understood to represent a value of infinity.
+
+### Colliders
+
+Rather than using a node's render mesh to perform collision detection, collision geometry must be explicitly declared by adding an additional `collider` property to a node.
+
+The `collider` property supplies three fields. The `shape` field describes the geometry which should be used to perform collision detection; the value indexes into the set of top-level collision shapes provided by the [KHR\_collision\_shapes](../KHR_collision_shapes/README.md) extension. The `physicsMaterial` indexes into the top-level set of physics materials, described in the "[Physics Materials](#physics-materials)" section of this document. Finally, the `collisionFilter` indexes into the top-level set of collision filters, described by the "[Collision Filtering](#collision-filtering)" section of this document.
+
+| |Type|Description|
+|-|-|-|
+|**shape**|`integer`| The index of a top-level `KHR_collision_shapes.shape`, which provides the geometry of the trigger.|
+|**physicsMaterial**|`integer`|Indexes into the top-level `physicsMaterials` and describes how the collider should respond to collisions.|
+|**collisionFilter**|`integer`|Indexes into the top-level `collisionFilters` and describes a filter which determines if this collider should perform collision detection against another collider.|
+
+If the node is part of a dynamic rigid body (i.e. itself or an ascendant has `motion` properties) then the collider belongs to that rigid body and must move with it during simulation. Otherwise the collider exists as a static object in the physics simulation which can be collided with but must not be moved as a result of solving collisions or joints. A rigid body may have a compound constructed from several collision shapes by supplying multiple descendent `collider` nodes.
+
+Implementations of this extension should ensure that collider transforms are always kept in sync with node transforms - for example animated node transforms should be applied to the physics engine (even for static colliders).
+
+
+#### Convex Hull Colliders
+
+Physics simulations typically recommend against allowing collisions between pairs of triangulated mesh objects, preferring to collide pairs of convex shapes instead. To support this, the `KHR_physics_rigid_bodies` extension may be defined on a `KHR_collision_shapes.shape` object.
+
+```javascript
+"extensions": {
+    "KHR_collision_shapes" : {
+        "shapes": [
+            {
+                "mesh": { "mesh": 0 },
+                "type": "mesh",
+                "extensions": {
+                    "KHR_physics_rigid_bodies": {
+                        "convexHull": true
```

Comment:
I see that this was removed from KHR_collision_shapes. Why? What are the intended users of KHR_collision_shapes that are not KHR_physics_rigid_bodies?

In [OMI_physics_gravity](https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_physics_gravity), I am making use of convex hulls, as it is one of the allowed shape types when the type of gravity is set to shaped gravity. It seems weird to put the definition of this inside of KHR_physics_rigid_bodies when it makes more sense to use KHR_collision_shapes.

Aside from KHR_physics_rigid_bodies and OMI_physics_gravity, which extensions need the collision shape types box, sphere, capsule, cylinder, and concave mesh, but not convex hull? Basically every physics system ever supports them. If an implementation does not support convex hulls for some reason, what use does it have for boxes and spheres, which are also convex objects? Or is the concern that people may misuse convex hulls somehow?

Reply:
KHR_collision_shapes is deliberately separated from KHR_physics_rigid_bodies to enable the geometries to be used by any number of future extensions, not necessarily limited to the domain of rigid body simulation. For example, node selection or 3D UI components, etc.

There was a concern that for these simpler use-cases, writing or importing code to generate convex hulls would be prohibitively complex. I agree that convex hulls are a necessity for rigid body simulation, which is why this property was moved to this extension. In addition, this configuration enables the rigid body specification to describe convex hulls of future shape types such as signed distance fields.